mono/metadata/threads.c

```diff
@@ -845,11 +846,7 @@ mono_thread_detach_internal (MonoInternalThread *thread)
 
 	mono_release_type_locks (thread);
 
-	/* Can happen when we attach the profiler helper thread in order to heapshot. */
-	if (!mono_thread_info_lookup (MONO_UINT_TO_NATIVE_THREAD_ID (thread->tid))->tools_thread)
```

Comment:
Interesting change, cuz the old code was incorrect.
MONO_PROFILER_RAISE uses hazard pointers in the same way as mono_thread_info_lookup, so whatever we need to be protected would be overwritten.

Reply:
Well, the callbacks invoked by `MONO_PROFILER_RAISE` may or may not use hazard pointers. The `mono_hazard_pointer_clear` below was there to clear HP1 (which would have been set by `mono_thread_info_lookup`) just in case the profiler callbacks wouldn't have overwritten it with something else. Was that approach actually problematic?